src/garage/torch/algos/trpo.py

```diff
@@ -0,0 +1,174 @@
+"""Trust Region Policy Optimization."""
+import torch
+
+from garage.torch.algos import VPG
+from garage.torch.optimizers import ConjugateGradientOptimizer
+from garage.torch.utils import flatten_batch
+
+
+class TRPO(VPG):
+    """Trust Region Policy Optimization (TRPO).
+
+    Args:
+        env_spec (garage.envs.EnvSpec): Environment specification.
+        policy (garage.torch.policies.base.Policy): Policy.
+        baseline (garage.np.baselines.Baseline): The baseline.
+        max_path_length (int): Maximum length of a single rollout.
+        policy_lr (float): Learning rate for training policy network.
+        n_samples (int): Number of train_once calls per epoch.
+        discount (float): Discount.
+        gae_lambda (float): Lambda used for generalized advantage
+            estimation.
+        max_kl (float): The maximum KL divergence between old and new
+            policies.
+        center_adv (bool): Whether to rescale the advantages
+            so that they have mean 0 and standard deviation 1.
+        positive_adv (bool): Whether to shift the advantages
+            so that they are always positive. When used in
+            conjunction with center_adv the advantages will be
+            standardized before shifting.
+        optimizer (object): The optimizer of the algorithm. Should be the
+            optimizers in torch.optim.
+        optimizer_args (dict): Arguments required to initialize the optimizer.
+        policy_ent_coeff (float): The coefficient of the policy entropy.
+            Setting it to zero would mean no entropy regularization.
+        use_softplus_entropy (bool): Whether to estimate the softmax
+            distribution of the entropy to prevent the entropy from being
+            negative.
+        stop_entropy_gradient (bool): Whether to stop the entropy gradient.
+        entropy_method (str): A string from: 'max', 'regularized',
+            'no_entropy'. The type of entropy method to use. 'max' adds the
+            dense entropy to the reward for each time step. 'regularized' adds
+            the mean entropy to the surrogate objective. See
+            https://arxiv.org/abs/1805.00909 for more details.
+
+    """
+
+    def __init__(self,
+                 env_spec,
+                 policy,
+                 baseline,
+                 max_path_length=500,
+                 policy_lr=3e-4,
+                 n_samples=1,
```

Comment:
n_samples? what does this do?

Reply:
please remove it if it's unused.

Reply:
It means "Number of train_once calls per epoch.". It gets passed down to its parent, and all the way up to `garage.np.Batchpoplot`, and used inside `train` function. I agree that the name is confusing and should be changed to something like `num_train_per_epoch`.

Whether we should remove this completely or not is a question for a separate PR I think.